src/main/java/com/google/devtools/build/lib/bazel/repository/RepoDefinitionValue.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.repository;
+
+import com.google.devtools.build.lib.cmdline.RepositoryName;
+import com.google.devtools.build.lib.skyframe.serialization.VisibleForSerialization;
+import com.google.devtools.build.lib.skyframe.serialization.autocodec.AutoCodec;
+import com.google.devtools.build.skyframe.AbstractSkyKey;
+import com.google.devtools.build.skyframe.NotComparableSkyValue;
+import com.google.devtools.build.skyframe.SkyFunctionName;
+import com.google.devtools.build.skyframe.SkyKey;
+
+/**
+ * The result of {@link RepoDefinitionFunction}, holding a repository rule instance.
+ *
+ * <p>This has to be a {@link NotComparableSkyValue} for a very subtle reason. Two {@link
```

Comment:
Could you add a paragraph about the consequences? As far as I can tell any change to a `SingleExtensionValue` will invalidate all the `RepoDefinitionValue`s, which will then invalidate all downstream values, including the actual repo directory, due to the lack of change pruning. But the marker file saves the day, so we would at most save some file reading and fingerprinting.

Reply:
is there anything in particular you want called out? What you said doesn't seem that surprising to me (and it was also the case before with `BzlmodRepoRuleValue`, btw, since `Package` did not compare equal).

Reply:
While this didn't really change, it was surprising to me when I first stumbled upon this two days ago :⁠-⁠) I thought I had found a major inefficiency until I remembered the marker file. Describing that this not being comparable isn't a big problem due to that file providing last minute change pruning would help.